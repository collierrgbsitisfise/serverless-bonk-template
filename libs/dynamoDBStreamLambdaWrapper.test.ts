import { DynamoDBStreamEvent, Context, Callback } from 'aws-lambda';

import { dynamoDBStreamLambdaWrapper } from './dynamoDBStreamLambdaWrapper';

describe('dynamoDBStreamLambdaWrapper', () => {
  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
  });

  const fakeLambdaResponse = {
    data: 'some data',
  };
  const fakeEvent = {} as DynamoDBStreamEvent;
  const fakeContext = {} as Context;
  const fakeCallBack = jest.fn() as Callback;

  it('should call "onSucces" handler', async () => {
    const lambda = jest.fn().mockReturnValueOnce(fakeLambdaResponse);
    const successHandler = jest.fn();
    const errorHandler = jest.fn();

    const executable = dynamoDBStreamLambdaWrapper(lambda, successHandler, errorHandler);
    expect.assertions(4);
    const res = await executable(fakeEvent, fakeContext, fakeCallBack);
    expect(res).toMatchSnapshot();
    expect(lambda).toBeCalledWith(fakeEvent, fakeContext, fakeCallBack);
    expect(successHandler).toBeCalledWith(fakeEvent, fakeLambdaResponse);
    expect(errorHandler).not.toBeCalledWith();
  });

  it('should call "onError" handler', async () => {
    const lambda = jest.fn().mockRejectedValueOnce(fakeLambdaResponse);
    const successHandler = jest.fn();
    const errorHandler = jest.fn();

    const executable = dynamoDBStreamLambdaWrapper(lambda, successHandler, errorHandler);
    expect.assertions(4);
    const res = await executable(fakeEvent, fakeContext, fakeCallBack);
    expect(res).toMatchSnapshot();
    expect(lambda).toBeCalledWith(fakeEvent, fakeContext, fakeCallBack);
    expect(successHandler).not.toBeCalledWith();
    expect(errorHandler).toBeCalledWith(fakeEvent, fakeLambdaResponse);
  });
});