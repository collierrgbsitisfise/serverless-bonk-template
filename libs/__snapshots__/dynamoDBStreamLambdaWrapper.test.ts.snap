// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamoDBStreamLambdaWrapper should call "onError" handler 1`] = `undefined`;

exports[`dynamoDBStreamLambdaWrapper should call "onSucces" handler 1`] = `undefined`;
